README.md

```diff
@@ -1,2 +1,90 @@
 # rosbot-xl-teleop-snap
-Building rosbot-xl-teleop snap
+
+The rosbot-xl teleop snap allows for teleoperating the robot from either the keyboard,
+a gamepad or leave it to the navigation stack.
+
+[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/rosbot-xl-teleop)
+
+## Installation
+
+Install the snap as follows,
+
+```bash
+snap install rosbot-xl-teleop
+```
+
+## Setup
+
+Upon installation, depending on your operating system,
+you may have to manually connect the snap interface.
+You can verify that with the following command,
+
+```bash
+$ snap connections rosbot-xl-teleop
+Interface            Plug                               Slot                            Notes
+content[ros-humble]  rosbot-xl-teleop:ros-humble        ros-humble-ros-base:ros-humble  manual
+hardware-observe     rosbot-xl-teleop:hardware-observe  -                               -
+joystick             rosbot-xl-teleop:joystick          -                               -
+network              rosbot-xl-teleop:network           :network                        -
+network-bind         rosbot-xl-teleop:network-bind      :network-bind                   -
+
+```
+
+The interface `ros-humble` must be connected.
+
+If it isn't, you can issue the following command,
+
+```bash
+snap connect rosbot-xl-teleop:ros-humble ros-humble-ros-base
```

Comment:
maybe the instruction to connect `hardware-observe` would be useful.

Reply:
I second this

Reply:
Done.